include/bluetooth/bluetooth.h

```diff
@@ -45,6 +45,64 @@ extern "C" {
  */
 #define BT_ID_DEFAULT 0
 
+/** Opaque type representing an advertiser. */
+struct bt_le_ext_adv;
+
+/* Don't require everyone to include conn.h */
+struct bt_conn;
+
+struct bt_le_adv_sent_info {
+	/* The number of advertising events completed. */
+	u8_t num_sent;
+};
+
+struct bt_le_adv_connected_info {
+	/* Connection object of the new connection */
+	struct bt_conn *conn;
+};
+
+struct bt_le_adv_scanned_info {
+	/* The Bluetooth device address of the active scanner */
+	bt_addr_le_t *addr;
+};
+
+struct bt_le_ext_adv_cb {
+	/** @brief The advertising set has finished sending adv data.
+	 *
+	 *  This callback notifies the application that the advertising set has
+	 *  finished sending advertising data.
+	 *  The advertising set can either have been stopped by a timeout or
+	 *  because the specified number of advertising events has been reached.
+	 *
+	 *  @param adv  The advertising set object.
+	 *  @param info Information about the sent event.
+	 */
+	void (*sent)(struct bt_le_ext_adv *adv,
+		     struct bt_le_adv_sent_info *info);
```

Comment:
What's the logic you're using to choose between `bt_le_ext_adv_*` and `bt_le_adv_*` as the name space? E.g. is `bt_le_adv_sent_info` used in anywhere else except for a callback inside a struct called `bt_le_ext_adv_cb`? I think it'd be good go get consistent here (or then I'm just not understanding what naming principle you've used). Ideally I'd have the "ext" part implicit (except where it needs to be explicit like when enabling a flag), but IIRC that was causing some conflicts with existing APIs?

Reply:
Faulty logic probably. Will update to include `_ext_`, these are new ones so shouldn't be a problem.

(Might be that in the beginning it was possible to have them for the old advertising API as well, but now it's only extended advertising API.)

Reply:
Found the logic: https://github.com/zephyrproject-rtos/zephyr/pull/22013#discussion_r375787031
Anyway I don't see a problem here. Using the new API will give you the ability to use the callbacks, without having  change the behavior of your advertiser.